packages/pn-pa-webapp/src/pages/__test__/NotificationDetail.page.test.tsx

```diff
@@ -27,119 +40,120 @@ jest.mock('react-i18next', () => ({
+    expect(NotificationDetailTimeline).toBeInTheDocument();
     // check payment history box
-    const paymentTable = result.getByTestId('paymentTable');
-    const paymentRecipient = result.getByTestId('paymentRecipient');
+    const paymentTable = result?.getByTestId('paymentTable');
     expect(paymentTable).toBeInTheDocument();
+    expect(paymentTable).toHaveTextContent(/2,00 €/i);
+    const paymentRecipient = result?.getByTestId('paymentRecipient');
     expect(paymentRecipient).toBeInTheDocument();
+
+    expect(mock.history.get).toHaveLength(2);
+    expect(mock.history.get[0].url).toContain('/notifications/sent');
+    expect(mock.history.get[1].url).toContain('/downtime/v1/history');
   });
 
-  test('executes the document and legal fact download handler', async () => {
-    const documentButton = result.getAllByTestId('documentButton');
-    const legalFactButton = result.getByTestId('legalFactButton');
-    expect(mockDispatchFn).toBeCalledTimes(1);
-    fireEvent.click(documentButton[0]);
-    expect(mockDispatchFn).toBeCalledTimes(2);
-    fireEvent.click(legalFactButton);
-    expect(mockDispatchFn).toBeCalledTimes(4);
+  it('executes the document and legal fact download handler', async () => {
+    mock.onGet(NOTIFICATION_DETAIL(notificationDTO.iun)).reply(200, notificationDTO);
+    mock.onGet(DOWNTIME_LEGAL_FACT_DETAILS(mockLegalIds.key)).reply(200, {
+      result: [],
+      nextPage: null,
+    });
+    mock.onGet(NOTIFICATION_DETAIL_DOCUMENTS(notificationToFe.iun, '0')).reply(200, {
+      filename: notificationToFe.documents[0].ref.key,
+      contentType: notificationToFe.documents[0].contentType,
+      contentLength: 3028,
+      sha256: notificationToFe.documents[0].digests.sha256,
+      url: 'https://mocked-url.com',
+    });
+    mock
+      .onGet(NOTIFICATION_DETAIL_LEGALFACT(notificationToFe.iun, mockLegalIds as LegalFactId))
+      .reply(200, {
+        filename: 'mocked-filename',
+        contentLength: 1000,
+        retryAfter: null,
+        url: 'https://mocked-url-com',
+      });
+    await act(async () => {
+      result = render(<NotificationDetail />);
+    });
+    const documentButton = result?.getAllByTestId('documentButton');
+    const legalFactButton = result?.getAllByTestId('download-legalfact');
+    await waitFor(() => fireEvent.click(documentButton![0]));
+    await waitFor(() => fireEvent.click(legalFactButton![0]));
+    expect(mock.history.get).toHaveLength(4);
+    expect(mock.history.get[0].url).toContain('/notifications/sent');
+    expect(mock.history.get[1].url).toContain('/downtime/v1/history');
+    expect(mock.history.get[2].url).toContain(
+      `/delivery/notifications/sent/${notificationToFe.iun}/attachments/documents/0`
+    );
+    expect(mock.history.get[3].url).toContain(
+      `/delivery-push/${notificationToFe.iun}/legal-facts/${mockLegalIds.category}/${mockLegalIds.key}`
+    );
```

Comment:
it's missed the check on the download of the legal fact linked to a downtime event